compiler/passes/codegen.cpp

```diff
@@ -1489,9 +1489,12 @@ void codegen(void) {
       USR_WARN("C code generation for packed pointers not supported");
   }
 
-  if(fIncrementalCompilation && fFastFlag)
-    USR_WARN("Compiling with --incremental and --fast together can lead to "
-             "slower execution time than compiling with --fast only");
+  std::size_t optimizationsEnabled = ccflags.find("-O");
+  if(fIncrementalCompilation && ( fFastFlag||
+      optimizationsEnabled!=std::string::npos ))
```

Comment:
Ooooooh.  I had completely forgotten that was a flag.  Sorry, I should've pinged you before merging